src/ynEIGEN/TokenStakingNode.sol

```diff
@@ -635,32 +560,41 @@ contract TokenStakingNode is ITokenStakingNode, Initializable, ReentrancyGuardUp
         IDelegationManagerTypes.Withdrawal memory _withdrawal
     ) internal {
         bytes32 withdrawalRoot = _delegationManager.calculateWithdrawalRoot(_withdrawal);
+        WithdrawableShareInfo storage _withdrawableShareInfo = withdrawableShareInfo[withdrawalRoot];
 
         // If the withdrawal was queued before the upgrade, it is considered legacy.
-        if (!queuedAfterSlashingUpgrade[withdrawalRoot]) {
-            legacyQueuedShares[_strategy] -= _withdrawal.scaledShares[0];
-
+        if (!_withdrawableShareInfo.postELIP002SlashingUpgrade) {
+            preELIP002QueuedSharesAmount[_strategy] -= _withdrawal.scaledShares[0];
             return;
         }
 
+        uint256 withdrawableSharesOfWithdrawalRoot = _withdrawableShareInfo.withdrawableShares;
+
         // To detect if the queued shares have not been synchronized after a slashing event, we compare the
         // maxMagnitude of the withdrawal root at the time of queueing with the current maxMagnitude.
-        uint64 maxMagnitudeAtSync = maxMagnitudeByWithdrawalRoot[withdrawalRoot];
-        uint64 maxMagnitudeNow = _allocationManager.getMaxMagnitude(_withdrawal.delegatedTo, _strategy);
+       
+       
+        // Also, given that only 1 strategy was withdrawn, we can expect the withdrawable shares array to contain only 1 value.
+        (, uint256[] memory _singleWithdrawableShares) = _delegationManager.getQueuedWithdrawal(withdrawalRoot);
+        uint256 withdrawableShares = _singleWithdrawableShares[0];
 
         // If they are different, it means that the queued shares have not been synced. 
         // In this case, it reverts to prevent accounting issues with the queuedShares variable.
-        if (maxMagnitudeAtSync != maxMagnitudeNow) {
-            revert NotSyncedAfterSlashing(withdrawalRoot, maxMagnitudeAtSync, maxMagnitudeNow);
+        if (withdrawableSharesOfWithdrawalRoot != withdrawableShares) {
+            revert NotSyncedAfterSlashing(withdrawalRoot);
         }
 
         // Decreases the queued shares by the withdrawable amount.
         // This will net to 0 after all queued withdrawals are completed.
-        queuedShares[_strategy] -= withdrawableSharesByWithdrawalRoot[withdrawalRoot];
+        queuedShares[_strategy] -= withdrawableSharesOfWithdrawalRoot;
+        _withdrawableShareInfo.withdrawableShares = 0;
```

Comment:
You could delete the whole element (including the bool)

delete _withdrawalShareInfo;